presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -38,4 +61,680 @@ public static JsonGenerator createJsonGenerator(JsonFactory factory, SliceOutput
+        {
+            String baseType = type.getTypeSignature().getBase();
+            if (type instanceof UnknownType) {
+                return new MapObjectKeyReader() {
+                    @Override
+                    public String read(Block block, int position)
+                    {
+                        return null;
+                    }
+                };
+            }
+            else if (baseType.equals(StandardTypes.BOOLEAN)) {
+                return new MapObjectKeyReader() {
+                    @Override
+                    public String read(Block block, int position)
+                    {
+                        return type.getBoolean(block, position) ? "true" : "false";
+                    }
+                };
+            }
+            else if (baseType.equals(StandardTypes.TINYINT) ||
+                    baseType.equals(StandardTypes.SMALLINT) ||
+                    baseType.equals(StandardTypes.INTEGER) ||
+                    baseType.equals(StandardTypes.BIGINT)) {
+                return new MapObjectKeyReader() {
+                    @Override
+                    public String read(Block block, int position)
+                    {
+                        return valueOf(type.getLong(block, position));
+                    }
+                };
+            }
+            else if (baseType.equals(StandardTypes.REAL)) {
+                return new MapObjectKeyReader() {
+                    @Override
+                    public String read(Block block, int position)
+                    {
+                        return valueOf(intBitsToFloat((int) type.getLong(block, position)));
+                    }
+                };
+            }
+            else if (baseType.equals(StandardTypes.DOUBLE)) {
+                return new MapObjectKeyReader() {
+                    @Override
+                    public String read(Block block, int position)
+                    {
+                        return valueOf(type.getDouble(block, position));
+                    }
+                };
+            }
+            else if (baseType.equals(StandardTypes.DECIMAL)) {
+                DecimalType decimalType = (DecimalType) type;
+                if (isShortDecimal(decimalType)) {
+                    return new MapObjectKeyReader() {
+                        @Override
+                        public String read(Block block, int position)
+                        {
+                            return Decimals.toString(decimalType.getLong(block, position), decimalType.getScale());
+                        }
+                    };
+                }
+                else {
+                    return new MapObjectKeyReader()
+                    {
+                        @Override
+                        public String read(Block block, int position)
+                        {
+                            return valueOf(new BigDecimal(
```

Comment:
@haozhun : Great point! `DecimalCasts` should probably be uniformed to use this method as well!